core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1313,6 +1313,11 @@ public static int localTime(DataContext root) {
     return (int) (localTimestamp(root) % DateTimeUtils.MILLIS_PER_DAY);
   }
 
+  /** SQL TRANSLATE(string, search_chars, replacement_chars) function. */
+  public static String translate3(String s, String search, String replacement) {
+    return org.apache.commons.lang3.StringUtils.replaceChars(s, search, replacement);
```

Comment:
Might be worthwhile to implement this ourselves (or find an implementation which we already depend on) than pulling in another dependency to do this.